includes/Embed_Base.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * Class Embed_Block.
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories;
+
+use Google\Web_Stories\Model\Story;
+use Google\Web_Stories\Story_Renderer\Image;
+use Google\Web_Stories\Story_Renderer\Embed;
+use Google\Web_Stories\Traits\Assets;
+
+/**
+ * Embed block class.
+ */
+class Embed_Base {
+	use Assets;
+
+	/**
+	 * Player script handle.
+	 *
+	 * @var string
+	 */
+	const STORY_PLAYER_HANDLE = 'standalone-amp-story-player';
+
+	/**
+	 * Script handle.
+	 *
+	 * @var string
+	 */
+	const SCRIPT_HANDLE = 'web-stories-embed';
+
+
+	/**
+	 * Initializes the Web Stories embed block.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @return void
+	 */
+	public function init() {
```

Comment:
My concern with this method being in the `Embed_Base` class is that it's run twice. All the `wp_register_script` and `wp_register_style` calls here are therefore run multiple time.

Now, for `wp_register_script` it's technically not a problem because it bails early if a script is already registered. However, I think we should not be doing that.

If we ever accidentally add anything here that should _not_ run twice (like the `wp_localize_script` call below), it might have unintended effects.

As such, I think registering of common scripts like `amp-story-player` should be moved out to `Plugin` or better yet, its own class.

Imagine for example turning the `Assets` trait into an actual class that _registers_ these scripts, and then the respective classes _enqueue_ them when needed.

Reply:
Fix in latest commit.